perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourcesManagerImpl.java

```diff
@@ -207,16 +206,16 @@ private void updateExtSource(PerunSession sess, ExtSource extSource) throws Inte
 		}
 
 		// Check the attributes
-		if (!extSourceDb.getAttributes().equals(extSource.getAttributes())) {
+		if (!getAttributes(extSourceDb).equals(attributes)) {
 			log.debug("There is a change in the extSource attributes");
-			try {
+				try {
 				// Firstly delete all attributes, then store new ones
 				jdbc.update("delete from ext_sources_attributes where ext_sources_id=?", extSource.getId());
 
-				for (String attrName: extSource.getAttributes().keySet()) {
+				for (String attrName: getAttributes(extSource).keySet()) {
```

Comment:
This is not correct. You already deleted all attributes from DB. Now you need to add new attributes from map in parameter of method.